scipy/stats/_distribution_infrastructure.py

```diff
@@ -2348,7 +2348,7 @@ def _median_formula(self, **params):
         raise NotImplementedError(self._not_implemented)
 
     def _median_icdf(self, **params):
-        return self._icdf_dispatch(0.5, **params)
+        return self._icdf_dispatch(np.asarray(0.5, dtype=self._dtype), **params)
```

Comment:
This `float` here was ultimately being passed to the `_ppf` method of `truncpareto`. Even though this would no longer be a problem due to the update to `truncpareto._ppf`, I thought I should fix it.